curriculum/challenges/english/07-scientific-computing-with-python/learn-encapsulation-by-building-a-projectile-trajectory-calculator/665d949e628d9f3fe305b6e8.md

```diff
@@ -0,0 +1,154 @@
+---
+id: 665d949e628d9f3fe305b6e8
+title: Step 17
+challengeType: 20
+dashedName: step-17
+---
+
+# --description--
+
+The output is becoming quite difficult to read, so you should replace the `print(graph.create_trajectory())` with a loop that prints each row separately. 
+
+# --hints--
+
+Remove the `print(graph.create_trajectory())` call.
+
+```js
+({
+    test: () => {
+        runPython(`
+prnt = _Node(_code).find_calls('print')
+assert not any(p.is_equivalent('print(graph.create_trajectory())') for p in prnt), "print still present"
+        `)
+    }
+})
+```
+
+You should have a loop that iterates over `graph.create_trajectory()`.
+
+```js
+({
+    test: () => {
+        runPython(`
+iter = _Node(_code).find_for_loops()[0].find_for_iter()
+assert iter.is_equivalent('graph.create_trajectory()'), "loop is missing"
+        `)
+    }
+})
+```
+
```

Comment:
````suggestion

You should print each row in `graph.create_trajectory()`.

```js
({ test: () => runPython(`
loop = _Node(_code).find_for_loops()[0]
var = str(loop.find_for_vars())
assert loop.find_bodies()[0].has_call(f"print({var})")
`) })
```

````
Something like this should work.